src/router/login_router.rs

```diff
@@ -1,24 +1,44 @@
-use rocket::{serde::json::Json, Route};
+use crate::{
+    db::{Db, GetUser},
+    jwt,
+    macros::{call, ToRocket},
+    model::Jwt,
+};
+use rocket::{http::Status, response::status, serde::json::Json, Route};
 use serde::{Deserialize, Serialize};
 
 pub fn routes() -> Vec<Route> {
     routes![login]
 }
 
-#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[derive(Serialize, Deserialize, Debug)]
 pub struct Login {
     pub email: String,
     pub password: String,
 }
 
-#[derive(Serialize, Deserialize, Debug, PartialEq)]
-pub struct Jwt {
-    pub jwt: String,
-}
-
 #[post("/", data = "<data>")]
-fn login(data: Json<Login>) -> Json<Jwt> {
-    Json(Jwt {
-        jwt: format!("{}:{}", data.email, data.password),
-    })
+async fn login(data: Json<Login>) -> Result<Json<Jwt>, status::Custom<String>> {
+    let user = call!(Db.GetUser::ByEmail(data.0.email))
+        .to_rocket("Failed to get user", Status::BadRequest)?;
```

Comment:
This looks more like a server error than a client error.
Maybe `InternalServerError`?

Reply:
Agree. However, this flaw exists all over the code base. Therefore, I would prefer to solve it in a separate PR

Reply:
Agreed